staging/src/k8s.io/code-generator/cmd/validation-gen/validation.go

```diff
@@ -683,35 +692,32 @@ func (g *genValidations) emitRegisterFunction(c *generator.Context, schemeRegist
 		}
 
 		targs := generator.Args{
-			"rootType":           t,
-			"statusType":         statusType,
-			"statusField":        statusField,
-			"fieldErrorList":     c.Universe.Type(fieldErrorListType),
-			"fieldInternalError": c.Universe.Type(fieldInternalErrorFunc),
-			"fieldPath":          c.Universe.Type(fieldPathType),
-			"fmtErrorf":          c.Universe.Type(fmtErrorfFunc),
-			"operationContext":   c.Universe.Type(operationContextType),
-			"safeCast":           c.Universe.Type(safeCastFunc),
-			"safeField":          c.Universe.Type(safeFieldFunc),
+			"rootType":    t,
+			"statusType":  statusType,
+			"statusField": statusField,
+			"field":       mkSymbolArgs(c, fieldPkgSymbols),
+			"fmt":         mkSymbolArgs(c, fmtPkgSymbols),
+			"operation":   mkSymbolArgs(c, operationPkgSymbols),
+			"safe":        mkSymbolArgs(c, safePkgSymbols),
 		}
 		// This uses a typed nil pointer, rather than a real instance because
 		// we need the type information, but not an instance of the type.
-		sw.Do("scheme.AddValidationFunc((*$.rootType|raw$)(nil), func(opCtx $.operationContext|raw$, obj, oldObj interface{}, subresources ...string) $.fieldErrorList|raw$ {\n", targs)
+		sw.Do("scheme.AddValidationFunc((*$.rootType|raw$)(nil), func(opCtx $.operation.Context|raw$, obj, oldObj interface{}, subresources ...string) $.field.ErrorList|raw$ {\n", targs)
```

Comment:
I like it.

Not for this PR, but I wonder if we can take this farther?  The "raw" namer has always felt weird to me in these contexts.  I'll make a note to follow up.

Reply:
if we renamed "raw" to "go" or "pkg" or "import" or something, would that clarify?  or `$.sym.operation.Context` for all symbols?

Reply:
I'll follow up.  I need to decide what scope I'm going for since this is in all our generators.